### ISSUE TITLE
Switch to closed-form Jacobian

### ISSUE THREAD

**Issue by Hoeze** [theislab/batchglm#46]
@davidsebfischer The gradient has to be adjusted for:
- batch_trainers
- batch_trainers_a_only
- batch_trainers_b_only
- full_data_trainers
- full_data_trainers_a_only
- full_data_trainers_b_only

**It has to be changed among others at the following lines:**
https://github.com/theislab/batchglm/blob/947e8b08086d5e10b8d9bc921939e1d7d940d4e4/batchglm/train/tf/nb_glm/estimator.py#L321-L322
https://github.com/theislab/batchglm/blob/947e8b08086d5e10b8d9bc921939e1d7d940d4e4/batchglm/train/tf/nb_glm/estimator.py#L364-L365
to `gradient = [..]` like in the following part:
https://github.com/theislab/batchglm/blob/947e8b08086d5e10b8d9bc921939e1d7d940d4e4/batchglm/train/tf/nb_glm/estimator.py#L328-L336

**Comment by davidsebfischer** [comment]
Gradients and Hessian both have a separate interface now and feed relevant tensors into MultiTrainer. Both can set to use analytic forms or tensorflow computed ones. Done in https://github.com/theislab/batchglm/pull/55.
